main/utils/errors.js

```diff
@@ -1,34 +1,161 @@
+const got = require('got');
+const delay = require('delay');
 
-const showError = async (error, {title, reportToSentry} = {}) => {
+const {showDialog} = require('../dialog');
+
+const MAX_RETRIES = 20;
+
+const getSentryIssue = async (eventId, tries = 0) => {
+  if (tries > MAX_RETRIES) {
+    return;
+  }
+
+  try {
+    const {body} = await got.get(`https://kap-sentry-tracker.vercel.app/api/event/${eventId}`, {json: true});
+
+    if (body.pending) {
+      await delay(1000);
+      return getSentryIssue(eventId, tries + 1);
+    }
+
+    return body;
+  } catch {
+
+  }
+};
+
+const getIssueBody = (title, errorStack, sentryTemplate) => `
+${sentryTemplate}<!--
+Thank you for helping us test Kap. Your feedback helps us make Kap better for everyone!
+-->
+
+**macOS version:**    ${os.release()} (darwin)
+**Kap version:**      ${app.getVersion()}
+\`\`\`
+${title}
+
+${errorStack}
+\`\`\`
+
+<!-- If you have additional information, enter it below. -->
+`;
+
+const showError = async (error, {title: customTitle, plugin} = {}) => {
   const ensuredError = ensureError(error);
-  const errorTitle = title || ensuredError.name;
+  const title = customTitle || ensuredError.name;
+  const detail = cleanStack(ensuredError.stack, {pretty: true});
 
-  console.error(error);
-  if (reportToSentry) {
-    // Imported here to avoid circular dependency
-    const Sentry = require('./sentry');
-    Sentry.captureException(ensuredError);
+  const mainButtons = [
+    'OK',
+    {
+      label: 'Copy Error',
+      action: () => clipboard.writeText(`${title}\n${cleanStack(ensuredError.stack)}`)
+    }
+  ];
+
+  // If it's a plugin error, offer to open an issue on the plugin repo (if known)
+  if (plugin) {
+    const openIssueButton = plugin.repoUrl && {
+      label: 'Open Issue',
+      action: () => {
+        openNewGitHubIssue({
+          repoUrl: plugin.repoUrl,
+          title,
+          body: getIssueBody(title, detail)
+        });
+      }
+    };
+
+    return showDialog({
+      title,
+      detail,
+      cancelId: 0,
+      defaultId: openIssueButton ? 2 : 0,
+      buttons: [...mainButtons, openIssueButton].filter(Boolean)
+    });
   }
 
-  // This is not currently really async: https://github.com/electron/electron/issues/23319
-  const {response} = await dialog.showMessageBox({
-    type: 'error',
-    message: errorTitle,
-    detail: cleanStack(ensuredError.stack, {pretty: true}),
-    buttons: [
-      'OK',
-      'Copy Error'
-    ]
-  });
+  // Avoids circular dependency
+  const settings = require('../common/settings');
+  const Sentry = require('./sentry');
+
+  let message;
+  const buttons = [...mainButtons];
+
+  if (isOnline && settings.get('allowAnalytics')) {
+    const eventId = Sentry.captureException(ensuredError);
+    const sentryIssuePromise = getSentryIssue(eventId);
+
+    message = 'Reporting this issue will help us track it better and resolve it faster';
+
+    buttons.push({
+      label: 'Collect info and report',
```

Comment:
```suggestion
      label: 'Collect Info and Report',
```

Reply:
We need to mention above what kind of information it will collect. Otherwise, I doubt anyone will want to click it.

Reply:
How do you think it should be worded? All we're doing is pretty much waiting for Sentry to categorize the event under an Issue and then looking for matching GH issues